src/lightning/fabric/connector.py

```diff
@@ -439,9 +439,11 @@ def _check_and_init_precision(self) -> Precision:
                         " is not supported with TPUs. Using `precision='bf16-mixed'` instead."
                     )
                 return XLABf16Precision()
+
         if isinstance(self.strategy, DeepSpeedStrategy):
             return DeepSpeedPrecision(self._precision_input)  # type: ignore
-
+        if isinstance(self.strategy, FSDPStrategy):
+            return FSDPPrecision(precision=self._precision_input)  # type: ignore[arg-type]
```

Comment:
Note: FSDP is not supported on CPU, so I removed the device argument (previously inherited from MixedPrecisionPlugin). Need to figure out how to handle backward compatibility.